fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java

```diff
@@ -96,5 +106,80 @@ public Void visitExecuteAsStatement(ExecuteAsStmt stmt, ConnectContext session)
             return null;
         }
 
+        @Override
+        public Void visitAlterUserStatement(AlterUserStmt stmt, ConnectContext session) {
+            analyseUserAndCheckExist(stmt.getUserIdent(), session);
+            /*
+             * IDENTIFIED BY
+             */
+            // convert password to hashed password
+            byte[] scramblePassword;
+            if (!Strings.isNullOrEmpty(stmt.getOriginalPassword())) {
+                try {
+                    // plain password should check for validation & reuse
+                    Auth.validatePassword(stmt.getOriginalPassword());
+                    GlobalStateMgr.getCurrentState().getAuth()
+                            .checkPasswordReuse(stmt.getUserIdent(), stmt.getOriginalPassword());
+                    if (stmt.isPasswordPlain()) {
+                        // convert plain password to scramble
+                        scramblePassword = MysqlPassword.makeScrambledPassword(stmt.getOriginalPassword());
+                    } else {
+                        scramblePassword = MysqlPassword.checkPassword(stmt.getOriginalPassword());
+                    }
+                } catch (DdlException | AnalysisException e) {
+                    // TODO AnalysisException used to raise in all old methods is captured and translated to SemanticException
+                    // that is permitted to throw during analyzing phrase under the new framework for compatibility.
+                    // Remove it after all old methods migrate to the new framework
+                    throw new SemanticException(e.getMessage());
+                }
+            } else {
+                scramblePassword = new byte[0];
+            }
+            stmt.setScramblePassword(scramblePassword);
+
+            /*
+             * IDENTIFIED WITH
+             */
+            if (!Strings.isNullOrEmpty(stmt.getAuthPlugin())) {
+                stmt.setAuthPlugin(stmt.getAuthPlugin().toUpperCase());
```

Comment:
It's a bit strange to read a field by getter, do some minor changes, and write it back at once by a setter all in one line. I suggest you put this logic inside the constructor of AlterUserStmt().

Reply:
nice suggestion, I move this to AstBuilder